### ISSUE TITLE
Procdump not working properly in STF

### ISSUE THREAD

**Issue by Mesbah-Alam** [AdoptOpenJDK/stf#86]
Procdump is used to create diagnostics, but it seems not working, the stdout shows the following: 

```
No process matching the specified PID can be found.
Try elevating the command prompt or using PsExec to make one as SYSTEM.
	psexec.exe -s -d -i cmd.exe
	procdump.exe -accepteula ...
```

Excerpt from test output: 

```
STF 11:32:46.269 - Using procdump.exe to generate .DMP files
STF 11:32:46.269 - Running command: F:\Users\jenkins\workspace\Test_openjdk8_j9_special.system_x86-32_windows_Personal_lambdaLoadTest\jvmtest\system\systemtest_prereqs\windows_sysinternals\procdump.exe -accepteula -ma 240
```

Details can be found from the example of a hanging lambda load test from internal build, output of which is attached here: 

[LambdaLoadTest_OpenJ9_NonLinux_special_24.zip](https://github.com/AdoptOpenJDK/stf/files/5453208/LambdaLoadTest_OpenJ9_NonLinux_special_24.zip)

**Comment by Mesbah-Alam** [comment]
[LambdaLoadTest_OpenJ9_NonLinux_special_24.zip](https://github.com/AdoptOpenJDK/stf/files/5453175/LambdaLoadTest_OpenJ9_NonLinux_special_24.zip)

**Comment by Mesbah-Alam** [comment]
Initial observations: 

In STF, procdump is invoked from [generate_process_diagnostics](https://github.com/AdoptOpenJDK/stf/blob/master/stf.core/scripts/stf/Commands.pm#L960) perl sub-routine. 

From the error message, it appears that the PID might be wrong. However, looking at the code above, the PID is being obtained via `$process->{pid}`. And the `else` block (for non-Windows case) is also using the same `$process` parameter -- which-- from instances seen in builds, is working fine, i.e., the script is "Sending SIG 3 to the java process to generate a javacore" as designed.

**Comment by Mesbah-Alam** [comment]
@lumpfish - could you please share your thoughts on the above? Is Cygwin somehow throwing us off here?

**Comment by lumpfish** [comment]
Do you know if it fails all the time or just on certain machines?  The example you have here is running 32 bit jdk - there are psexec64.exe and procdump64.exe for 64 bit processes.  Do you know if it also fails on 64 bit?

**Comment by lumpfish** [comment]
Trying to run psexec.exe on a local machine (not as Administrator) I get the following:

1 `psexec.exe -s -d -i cmd.exe`.  I receive a pop-up saying that I must accept the enduser license agreement.
2. `psexec.exe -accepteula -s -d -i cmd.exe` (i.e. accept the eula on the command line), I see:
```
C:\Users\srushton\systemtest_prereqs\windows_sysinternals>psexec.exe -s -d -i cmd.exe

PsExec v2.2 - Execute processes remotely
Copyright (C) 2001-2016 Mark Russinovich
Sysinternals - www.sysinternals.com

Couldn't install PSEXESVC service:
Access is denied.
```
So it may be that the user running the tests in the Jenkins pipelines is not able to execute the program, or needs to accept the eula on the command line.

**Comment by lumpfish** [comment]
Some more investigation and I don't think psexec permissions has anything to do with it - psexec is not being run in the test scripts at al).  I've reproduced the symptoms locally - the pid returned from win32::CreateProcess does not seem to be the same as the pid procdump is expecting.  Trying to work out what's going on.......

**Comment by lumpfish** [comment]
Turns out this is due to 64 vs. 32 bit perl.
The test script starts a process using win32::CreateProcess and then extracts the pid from the returned ProcessInformation structure.
Preceding the pid in the structure are two pointers - these are assumed to be 4 bytes and the pid therefore starts at byte 9.
When 64 bit perl is used the pointers are 8 bytes and the pid starts at byte 17.  So the pid being passed to procdump (and to other programs / functions such as StopProcess) are not correct.
There is a fix in https://github.com/AdoptOpenJDK/stf/pull/89 which asks perl what the size of a pointer is in the perl running the test and extracts the pid accordingly.
I've also included a new 'SampleOverrunProcess' test case which can be used to test that hung processes are being dumped and killed as expected.

The sample can be run via
```
make test.SampleOverrunProcess
```
(need to build the project first - see https://github.com/AdoptOpenJDK/stf/blob/master/README.md)

sanity.system test runs are in
jdk8
openj9
x86-64_windows: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4541/ (two test failures but unrelated to this change)
jdk11
hotspot
x86-64_windows: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4542/
